.github/workflows/assigner.yml

```diff
@@ -24,23 +24,32 @@ jobs:
     if: github.event.pull_request.draft == false
     runs-on: ubuntu-24.04
     permissions:
-      pull-requests: write # to add assignees to pull requests
       issues: write        # to add assignees to issues
 
     steps:
     - name: Check out source code
       uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+      with:
+        fetch-depth: 0
+        persist-credentials: false
 
     - name: Set up Python
-      uses: actions/setup-python@a26af69be951a213d495a4c3e4e4022e16d87065 # v5.6.0
+      uses: zephyrproject-rtos/action-python-env@ace91a63fd503cd618ff1eb83fbcf302dabd7d44 # main
       with:
         python-version: 3.12
-        cache: pip
-        cache-dependency-path: scripts/requirements-actions.txt
 
-    - name: Install Python packages
+    - name: Fetch west.yml from pull request
+      run: |
+        git fetch origin pull/${{ github.event.pull_request.number }}/head
+        git show FETCH_HEAD:west.yml > pr_west.yml
+
+    - name: west setup
+      if: >
+        github.event_name == 'pull_request_target'
       run: |
-        pip install -r scripts/requirements-actions.txt --require-hashes
+        git config --global user.email "you@example.com"
+        git config --global user.name "Your Name"
+        west init -l . || true
```

Comment:
why the `|| true`? do you expect west init to fail? Guess some scenario where there is a workspace initialized already but it feels odd